### PR TITLE
Thumbnail mode

### LINKED ISSUE #88: Feature request: image thumbnails
In order to display image thumbnails, I will create a different view specific to images

This mode will be used when the API is called like this:

```
ce.openFile(['.jpg', '.jpeg', '.png', '.gif'])
.then(fileInfo => {
    ...
})
```
This means these tasks

* [x] new "mode" for images, with thumbnails list
* [x] new design
* [x] resize images on the fly

If anyone has ideas (design or code) please comment :)
cc @clemos @ceubri @suddenheroes

### PR DESCRIPTION
**Pull request by lexoyo**
screenshot_20191024_104323
screenshot_20191024_103314

closes #88

### CODE REVIEW

**Comment by lexoyo** [comment]
Before merging this I need 
* [ ] remove the env var and make it an option passed to the router like any other option

**Comment by JbIPS** [comment]
I'll look at it monday

**Comment by lexoyo** [comment]
Merci !

File: lib/router.js [hunk 1]
```diff
@@ -17,6 +17,11 @@ const multer = require('multer');
 const storage = multer.memoryStorage();
 const upload = multer({storage});
 
+// images
+const sizeOf = require('buffer-image-size');
+const THUMBNAIL_SIZE = 256;
```
[the remark this answers is not in the document]
Comment (lexoyo) [reply lib/router.js]: Nice catch !

File: lib/router.js [hunk 2]
```diff
@@ -164,15 +176,53 @@ module.exports = class Router extends express.Router {
       .catch((err) => Router.handleError(res, err));
     });
 
+    const iconsRoot = Path.resolve(nodeModulesPath('file-icon-vectors'), 'file-icon-vectors/dist/icons/square-o');
+    const catalog = require(iconsRoot + '/catalog.json');
+    this.get(/\/(.*)\/icon\/(.*)/, withCache, (req, res) => {
+      const service = req.params['0'];
+      const file = req.params['1'];
+      const ext = file.split('.').pop().split('?').shift().toLowerCase();
+      if(['jpg', 'jpeg', 'png'].includes(ext)) {
```
[the remark this answers is not in the document]
Comment (lexoyo) [reply lib/router.js]: :+1:

File: README.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ Manage your users' cloud services from your application.
 
 ## Install
 
+[Image Magick](https://github.com/ImageMagick/ImageMagick) is required unless you set the env var `ENABLE_IMAGE_MAGICK=false`, which will disable the resize of images for thumbnails (in this case it is discouraged to use the thumbnail mode).
```
[the remark this answers is not in the document]
Comment (lexoyo) [reply README.md]: I tried sharp first and I don't remember why I went for imagemagik
I'll give it another try

[the remark this answers is not in the document]
Comment (lexoyo) [reply README.md]: Thank you very much @JbIPS for this review
I fixed everything and sharp worked like a charm !

**Pull request merged**